controller/work_item_boardcolumns_blackbox_test.go

```diff
@@ -0,0 +1,257 @@
+		}
+		// when
+		_, updatedWI := test.UpdateWorkitemOK(t, svc.Context, svc, ctrl, fxt.WorkItems[0].ID, &u)
+		// then
+		assert.NotNil(t, updatedWI)
+		assert.Len(t, updatedWI.Data.Relationships.SystemBoardcolumns.Data, 2)
+		mustHave := map[string]struct{}{
+			*u.Data.Relationships.SystemBoardcolumns.Data[0].ID: {},
+			*u.Data.Relationships.SystemBoardcolumns.Data[2].ID: {},
+		}
+		for _, lblData := range updatedWI.Data.Relationships.SystemBoardcolumns.Data {
+			delete(mustHave, *lblData.ID)
+		}
+		require.Empty(t, mustHave)
+	})
+}
+
+func (l *TestWorkItemBoardcolumnREST) TestRemoveAllBoardcolumns() {
+	wiCnt := 2
+	boardCnt := 2
+	fxt := tf.NewTestFixture(l.T(), l.DB, tf.CreateWorkItemEnvironment(), tf.WorkItems(wiCnt), tf.WorkItemBoards(boardCnt))
+	svc, ctrl := l.SecuredController(*fxt.Identities[0])
+
+	l.T().Run("Fetch WI and verify boardcolumns Relationship", func(t *testing.T) {
+		_, fetchedWI := test.ShowWorkitemOK(t, svc.Context, svc, ctrl, fxt.WorkItems[0].ID, nil, nil)
+		require.NotNil(t, fetchedWI.Data.Relationships.SystemBoardcolumns)
+		assert.Empty(t, fetchedWI.Data.Relationships.SystemBoardcolumns.Data)
+	})
+
+	l.T().Run("add a column reference", func(t *testing.T) {
+		// given
+		u := app.UpdateWorkitemPayload{
+			Data: &app.WorkItem{
+				ID:   &fxt.WorkItems[0].ID,
+				Type: APIStringTypeWorkItem,
+				Attributes: map[string]interface{}{
+					"version": fxt.WorkItems[0].Version,
+				},
+				Relationships: &app.WorkItemRelationships{
+					SystemBoardcolumns: &app.RelationGenericList{
+						Data: []*app.GenericData{
+							{
+								ID:   ptr.String(fxt.WorkItemBoards[0].Columns[0].ID.String()),
+								Type: ptr.String("boardcolumns"),
+							},
+							{
+								ID:   ptr.String(fxt.WorkItemBoards[1].Columns[1].ID.String()),
+								Type: ptr.String("boardcolumns"),
+							},
+						},
+					},
+				},
+			},
+		}
+		// when
+		_, updatedWI := test.UpdateWorkitemOK(t, svc.Context, svc, ctrl, fxt.WorkItems[0].ID, &u)
+		// then
+		assert.NotNil(t, updatedWI)
+		assert.Len(t, updatedWI.Data.Relationships.SystemBoardcolumns.Data, 2)
+		mustHave := map[string]struct{}{
+			*u.Data.Relationships.SystemBoardcolumns.Data[0].ID: {},
+			*u.Data.Relationships.SystemBoardcolumns.Data[1].ID: {},
+		}
+		for _, lblData := range updatedWI.Data.Relationships.SystemBoardcolumns.Data {
+			delete(mustHave, *lblData.ID)
+		}
+		require.Empty(t, mustHave)
+
+		t.Run("now remove all columns", func(t *testing.T) {
+			// given
+			u.Data.Attributes["version"] = updatedWI.Data.Attributes["version"]
+			u.Data.Relationships.SystemBoardcolumns = &app.RelationGenericList{
+				Data: []*app.GenericData{},
+			}
+			// when
+			_, updatedWI = test.UpdateWorkitemOK(t, svc.Context, svc, ctrl, fxt.WorkItems[0].ID, &u)
+			// then
+			assert.NotNil(t, updatedWI)
+			assert.Empty(t, updatedWI.Data.Relationships.SystemBoardcolumns.Data)
+		})
+	})
+}
+
+/* FIXME(michaelkleinhenz): Add tests as soon as isValid is added to workitem.go
```

Comment:
Should this FIXME be part of this PR?

Reply:
Yes. We have a convention here that TODOS and FIXMES are labeled with the GH username. FIXMES are usually things we already know we need to do at some point down the line but the precondition is not here yet.

Reply:
Can we make an issue out of it and remove this dead code? It can end up being confusing over the time.

Reply:
@bartoszmajsak there are a lot of `TODO`s in the code and they have a point IMHO. Sometimes it is about better performance or ideas for optimizations worth looking at.